pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageImpl.java

```diff
@@ -75,12 +81,8 @@ static MessageImpl create(MessageMetadata.Builder msgMetadataBuilder, ByteBuffer
         this.payload = Unpooled.copiedBuffer(payload);
 
         if (msgMetadata.getPropertiesCount() > 0) {
-            Map<String, String> properties = Maps.newTreeMap();
-            for (KeyValue entry : msgMetadata.getPropertiesList()) {
-                properties.put(entry.getKey(), entry.getValue());
-            }
-
-            this.properties = Collections.unmodifiableMap(properties);
+            this.properties = Collections.unmodifiableMap(msgMetadataBuilder.getPropertiesList().stream()
```

Comment:
I think this might also affect the replicator messages, in where we don't really care about the properties

Reply:
Maybe we could set properties to null and do a lazy population

Reply:
> I think this might also affect the replicator messages, , Maybe we could set properties to null and do a lazy population

Make sense. fixed it.